utils/writeDataFile.js

```diff
@@ -0,0 +1,31 @@
+const fs = require('fs');
+
+// iterates over a mongocursor and writes a json file with "data" and "metadata" key-value pairs
+const writeDataFile = (fileName, mongoCursor, metadata, callback) => {
+  // init writable stream
+  const file = fs.createWriteStream(fileName, { flags: 'a' });
+
+  // create opening lines of json file
+  file.write('{\n"data": {"actions":[\n');
+
+  // var separator is used in forEach loop below to delimit mongodb docs when writing to file
+  let separator = '';
+  mongoCursor.forEach(
+    (document) => {
+      file.write(separator + JSON.stringify(document, null, 2));
```

Comment:
Is this the "only/easy" way to write to files? Because: https://nodejs.org/api/stream.html#stream_writable_write_chunk_encoding_callback

Check the return value of `write`. Apparently you should not write the next chunk if it returns `false`(one should wait for `drain` event)

Reply:
here's my thinking:

1. [`fs.writeFile`](https://nodejs.org/api/fs.html#fs_fs_writefile_file_data_options_callback) does not make sense since it will overwrite a file that already exists - therefore sequential writes are not possible. this option was suitable when we used the `.toArray()` method on the cursor

2. `fs.appendFile` is problematic - see [here](https://stackoverflow.com/a/43370201) for an explanation why. in fact in some of my tests I ended up with corrupted files when using this function. this SO source recommends the `createWriteStream` approach I ended up using

3. `fs.write` is not suitable, either. From [docs](https://nodejs.org/api/fs.html#fs_fs_write_fd_buffer_offset_length_position_callback): '_It is unsafe to use fs.write() multiple times on the same file without waiting for the callback. For this scenario, fs.createWriteStream() is recommended._'

so what options are left? let me know if you have any thoughts, including potentially editing the createWriteStream approach I use